catalog/Catalog.proto

```diff
@@ -99,6 +99,7 @@ message CatalogRelation {
     optional IndexScheme index_scheme = 18;
     optional PartitionScheme partition_scheme = 19;
     optional NUMAPlacementScheme placement_scheme = 20;
+    optional uint32 num_tuples = 21;
```

Comment:
Also needs to update `CatalogRelation::getProto()` and `CatalogRelation(const serialization::CatalogRelationSchema &proto)` to serialize / deserialize this field.